File: app/src/main/java/com/mikhailovskii/trakttv/ui/movies_list/MovieListContract.java
package com.mikhailovskii.trakttv.ui.movies_list;

import androidx.annotation.NonNull;

import com.mikhailovskii.trakttv.data.entity.Movie;
import com.mikhailovskii.trakttv.ui.base.MvpPresenter;
import com.mikhailovskii.trakttv.ui.base.MvpView;

import java.util.List;

public interface MovieListContract {

    interface MoviesListView extends MvpView {

        void onMovieListLoaded(@NonNull List<Movie> movieList);

        void onMovieListFailed();

    }

    interface MoviesListPresenter extends MvpPresenter<MoviesListView> {

        void loadMovieList();

    }

}

File: app/src/main/java/com/mikhailovskii/trakttv/ui/movies_list/MovieListContract.kt
package com.mikhailovskii.trakttv.ui.movies_list

import com.mikhailovskii.trakttv.data.entity.Movie
import com.mikhailovskii.trakttv.ui.base.MvpPresenter
import com.mikhailovskii.trakttv.ui.base.MvpView

interface MovieListContract {

    interface MovieListView : MvpView {

        fun onMovieListLoaded(movieList: List<Movie>)

        fun onMovieListFailed()

    }

    interface MovieListPresenter : MvpPresenter<MovieListView> {

        fun loadMovieList()

    }

}
